Applications/xia2_main.py

```diff
@@ -63,6 +66,38 @@ def check_environment():
         )
 
 
+def check_hdf5_master_files(master_files):
+    """Check the input HDF5 master files look a little bit like HDF5 master
+    files and not just the data files: if the latter then sys.exit() with a
+    helpful message"""
+
+    bad = []
+
+    for filename in master_files:
+        with h5py.File(filename, "r") as f:
+            if b"/data" in f and b"/entry" not in f:
```

Comment:
Not sure about the wider context of this function. Do we already know that h5py can read these files or should we also catch any exceptions here?
(If in doubt: leave as is and sort out later)

Reply:
Guess it would be "bad" if `h5py` can't read so ... yeh will catch

Reply:
Catching `OSError` seems to get most everything